subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -519,6 +519,12 @@ void ll_reset(void)
 #endif /* !CONFIG_BT_CTLR_ZLI */
 	}
 
+#if defined(CONFIG_BT_BROADCASTER)
+	/* Init after adv state reset */
+	err = ull_adv_reset_init();
```

Comment:
`ull_adv_reset` is called above, why is this call to `ull_adv_reset_init` necessary here?

Reply:
The order of reset procedure is:
1. Reset ULL context, like scheduling, abort LLL events etc.
2. Reset LLL context, like event members etc.
3. Reset ULL static members (which otherwise was cleared if on power up)
4. Init ULL context members

Reply:
Can step 3. be done in step 1. (or vice versa)? If not, we should try to find better names, having ..._init, ..._reset, ..._reset_init and static functions called init_reset is confusing.

Reply:
I will some inline comments and find suitable names.

Reply:
I guess the above comment still needs to be fixed w.r.t. renaming the confusing `init_reset` function.

Reply:
I still don't understand why step 3 cannot be done as part of step 1. Can you please explain?

Reply:
step 3, cannot be done in step 1 because step 2 (call to `lll_adv_reset`) initializes the free AC PDU memory pool, which is needed in step 3.

You can suggest alternative names, current names reflect their purpose:
`ull_<role>_init` : Power up ULL initialization (Thread executable)
`lll_<role>_init`: Power up LLL initialization (Thread executable)
`ull_<role>_reset`: Warm reset ULL operation (Thread executable)
`lll_<role>_reset`: Warm reset LLL operation (LLL executable)
`ull_<role>_reset_init`: Initialization of ULL context after LLL warm reset operation (Thread executable)
`init_reset`: common code in `ull_<role>_init`, `ull_<role>_reset`, and `ull_<role>_reset_init` (Thread executable)

Reply:
I think the _init/_reset functions are appropriately named. I considered some alternative names for `ull_<role>_reset_init`; the clearest I can come up with is `ull_<role>_finalize_reset` (as in "finalizing the ULL reset after LLL has been reset"). Unless the other reviewers prefer `ull_<role>_reset_init`, I propose to go for that.

Reply:
Do you agree with `ull_<role>_reset_finalize`? I would like to keep the verb as suffix.

Reply:
That works for me.